src/app/common/modal/modal/modal.component.ts

```diff
@@ -1,29 +1,41 @@
-import { Component } from '@angular/core';
-
-import { BsModalRef } from 'ngx-bootstrap/modal';
-import { AbstractModalDirective } from '../abstract-modal.directive';
-import { ModalAction, ModalCloseEvent, ModalInput } from '../modal.model';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
 
 @Component({
-	templateUrl: 'modal.component.html'
+	selector: 'asy-modal',
+	templateUrl: './modal.component.html',
+	styleUrls: ['./modal.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ModalComponent extends AbstractModalDirective {
-	message: string;
+export class ModalComponent {
```

Comment:
@jrassa Given the amount of overlap between these inputs and the `abstract-modal.directive` properties, would it make sense to move the inputs into the abstract directive and have the `ModalComponent` extend `AbstractModalDirective`  with the added `@Output()`s?

In the `abstract-modal.directive` We could add `@Input()` decorators to `title`, `okText`, and `cancelText` fields, and add `disableOk` and `autoCaptureFocus` inputs. Those last two are used by the `ContainerModalComponent` too, although in that component they're called `isDisableOk` and `isCdkFocusInitial`.

Reply:
I debated about extending from `AbstractModalDirective`.  Ultimately I decided against because while there is some overlap there is also enough differences, that I worry would clutter the class and potentially cause confusion (i.e. ok/cancel event emitters vs. ok/cancel methods and onClose subject) over how to use.  As well, this modal class is just focused on eliminating duplicate markup and is decoupled from the actual modal implementation (ngx-bootstrap).

If there was duplicated logic, I would be more inclined to try and consolidate, but I don't think 3 properties is enough overlap to be concerned about.